src/Interfaces.jl

```diff
@@ -158,16 +193,24 @@ function toimplements!(IT, arg::Expr, shouldthrow::Bool=true)
             arg = arg.args[2]
         end
         arg.args[2] = toimplements!(IT, arg.args[2], false)
-        return :(($origarg) || Interfaces.atleastonerequired($IT, $(Meta.quot(argcopy))))
+        return quote
+            check = $origarg
+            check || Interfaces.atleastonerequired(debuglvl, $IT, $(Meta.quot(argcopy)))
+            check
+        end
     elseif arg.head == :block
         # not expected at top-level of @interface block
         # but can be block of if-else or || expressions
         map!(x -> toimplements!(IT, x, shouldthrow), arg.args, arg.args)
         return arg
     elseif arg.head == :macrocall && arg.args[1] == Symbol("@optional")
         return quote
-            if @isdefined(debug) && debug
-                $(toimplements!(IT, arg.args[3]))
+            if debug
+                oldlvl = debuglvl
+                debuglvl = :warn
+                check = $(toimplements!(IT, arg.args[3], shouldthrow))
+                debuglvl = oldlvl
```

Comment:
this doesn't seem the most elegant to me... but it seemed better than any alternative I came up with (e.g. passing a `debuglvl` into `toimplements!`)

Reply:
Can you explain a little more why we need to store/reset the `debuglvl` here?

Reply:
we set `debuglvl` at the top of the `implements` method and then use this variable thoughout (similar to `mods`), so `toimplements!` has to be able to rely on this variable being defined (and defined correctly) no matter the order in which the expression in `@interface` appear. That's why, whenever `toimplements!` changes `debuglvl` is has to put it back again.


without this, an `@optional` declaration will leave set the `debuglvl` set

Here's an example of what happens if we don't reset it, given an interface with an `@optional` in the middle:
```jl
julia> @interface AbstractTable begin
           Tables.isrowtable(::Type{AbstractTable})

           @optional Tables.schema(::AbstractTable)::Union{Nothing, Tables.Schema}

           Tables.istable(::Type{AbstractTable})
       end
```
log messages after the `@optional` will be at the wrong level:
```jl
julia> struct Foo end

julia> Interfaces.implements(Foo, AbstractTable; debug=true)
[ InterfaceImplementationError: missing `AbstractTable` interface method definition: `(Tables.isrowtable)(::Type{Foo})`, in module(s): `Module[Main]`
[ @optional definition missing: missing `AbstractTable` interface method definition: `(Tables.schema)(::Foo)`, in module(s): `Module[Main]`
[ @optional definition missing: missing `AbstractTable` interface method definition: `(Tables.istable)(::Type{Foo})`, in module(s): `Module[Main]`
false
```

Open to alternative ideas for how we control logging level :)

Reply:
That makes sense to me; thanks.